### ISSUE TITLE
Sound does not play on client if started after application

### ISSUE THREAD

**Issue by Lombra** [LizardByte/Sunshine#3396]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the latest beta/pre-release?

This issue is present in the latest pre-release

### Describe the Bug

If I launch a game and then start streaming to a client, the game sound plays only on the host. Any application launched while a stream is active will get sent to the stream client, even after I've exited and reopened the stream.

### Host Operating System

Windows

### Operating System Version

Windows 10 build 19045.5011

### Architecture

amd64/x86_64

### Sunshine commit or version

v2024.1109.210203

### Package

Windows - portable (not recommended)

### GPU Type

Nvidia

### GPU Model

Geforce RTX 3080

### GPU Driver/Mesa Version

536.23

### Capture Method

None

### Relevant log output

```shell
[2024-11-13 22:05:24.015]: Info: Sunshine version: v2024.1109.210203
[2024-11-13 22:05:24.015]: Info: Package Publisher: LizardByte
[2024-11-13 22:05:24.015]: Info: Publisher Website: https://app.lizardbyte.dev
[2024-11-13 22:05:24.015]: Info: Get support: https://app.lizardbyte.dev/support
[2024-11-13 22:05:24.215]: Info: nvprefs: Opened undo file from previous improper termination
[2024-11-13 22:05:24.215]: Info: nvprefs: Restored OGL_CPL_PREFER_DXPRESENT for base profile
[2024-11-13 22:05:24.251]: Info: nvprefs: Restored global profile settings from undo file - deleting the file
[2024-11-13 22:05:24.253]: Info: nvprefs: No need to modify application profile settings
[2024-11-13 22:05:24.253]: Info: nvprefs: Changed OGL_CPL_PREFER_DXPRESENT to OGL_CPL_PREFER_DXPRESENT_PREFER_ENABLED for base profile
[2024-11-13 22:05:24.344]: Info: Compiling shaders...
[2024-11-13 22:05:24.385]: Info: System tray created
[2024-11-13 22:05:24.424]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-11-13 22:05:24.474]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-11-13 22:05:24.517]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-11-13 22:05:24.544]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-11-13 22:05:24.609]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-11-13 22:05:24.634]: Warning: include/base_vs.hlsl(28,13-36): warning X3556: integer modulus may be much slower, try using uints if possible.
[2024-11-13 22:05:24.634]: Info: Compiled shaders
[2024-11-13 22:05:24.811]: Info: Successfully reset default audio device
[2024-11-13 22:05:24.812]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024-11-13 22:05:24.812]: Info: Trying encoder [nvenc]
[2024-11-13 22:05:25.007]: Info: ddprobe.exe 1  --verify-frame-capture returned 0x00000000
[2024-11-13 22:05:25.007]: Info: Set GPU preference: 1
[2024-11-13 22:05:25.232]: Info:
Device Description : NVIDIA GeForce RTX 3080
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002216
Device Video Mem   : 10067 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1440
Offset             : 0x0
Virtual Desktop    : 2560x1440
[2024-11-13 22:05:25.233]: Info: Active GPU has HAGS disabled
[2024-11-13 22:05:25.233]: Info: Using realtime GPU priority
[2024-11-13 22:05:25.233]: Info:
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.685547,0.308594]
Green Primary      : [0.264648,0.667969]
Blue Primary       : [0.150391,0.0576172]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.2465 nits
Max Luminance      : 417.712 nits
Max Full Luminance : 417.712 nits
[2024-11-13 22:05:25.236]: Info: Desktop resolution [2560x1440]
[2024-11-13 22:05:25.237]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024-11-13 22:05:25.237]: Info: Display refresh rate [165.08Hz]
[2024-11-13 22:05:25.237]: Info: Requested frame rate [60fps]
[2024-11-13 22:05:25.237]: Info: Creating encoder [h264_nvenc]
[2024-11-13 22:05:25.237]: Info: Color coding: SDR (Rec. 601)
[2024-11-13 22:05:25.237]: Info: Color depth: 8-bit
[2024-11-13 22:05:25.237]: Info: Color range: JPEG
[2024-11-13 22:05:25.317]: Info: NvEnc: created encoder H.264 P1 two-pass rfi
[2024-11-13 22:05:25.389]: Info: Creating encoder [hevc_nvenc]
[2024-11-13 22:05:25.389]: Info: Color coding: SDR (Rec. 601)
[2024-11-13 22:05:25.390]: Info: Color depth: 8-bit
[2024-11-13 22:05:25.390]: Info: Color range: JPEG
[2024-11-13 22:05:25.480]: Info: NvEnc: created encoder HEVC P1 two-pass rfi
[2024-11-13 22:05:25.517]: Info: Creating encoder [av1_nvenc]
[2024-11-13 22:05:25.517]: Info: Color coding: SDR (Rec. 601)
[2024-11-13 22:05:25.517]: Info: Color depth: 8-bit
[2024-11-13 22:05:25.517]: Info: Color range: JPEG
[2024-11-13 22:05:25.558]: Error: NvEnc: encoding format is not supported by the gpu
[2024-11-13 22:05:25.566]: Info: Creating encoder [h264_nvenc]
[2024-11-13 22:05:25.567]: Info: Color coding: SDR (Rec. 601)
[2024-11-13 22:05:25.567]: Info: Color depth: 8-bit
[2024-11-13 22:05:25.567]: Info: Color range: JPEG
[2024-11-13 22:05:25.645]: Info: NvEnc: created encoder H.264 P1 yuv444 two-pass rfi
[2024-11-13 22:05:25.899]: Info:
Device Description : NVIDIA GeForce RTX 3080
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002216
Device Video Mem   : 10067 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1440
Offset             : 0x0
Virtual Desktop    : 2560x1440
[2024-11-13 22:05:25.899]: Info: Active GPU has HAGS disabled
[2024-11-13 22:05:25.900]: Info: Using realtime GPU priority
[2024-11-13 22:05:25.900]: Info:
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.685547,0.308594]
Green Primary      : [0.264648,0.667969]
Blue Primary       : [0.150391,0.0576172]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.2465 nits
Max Luminance      : 417.712 nits
Max Full Luminance : 417.712 nits
[2024-11-13 22:05:25.902]: Info: Desktop resolution [2560x1440]
[2024-11-13 22:05:25.903]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024-11-13 22:05:25.903]: Info: Display refresh rate [165.08Hz]
[2024-11-13 22:05:25.903]: Info: Requested frame rate [60fps]
[2024-11-13 22:05:25.903]: Info: Creating encoder [hevc_nvenc]
[2024-11-13 22:05:25.903]: Info: Color coding: SDR (Rec. 709)
[2024-11-13 22:05:25.903]: Info: Color depth: 10-bit
[2024-11-13 22:05:25.903]: Info: Color range: JPEG
[2024-11-13 22:05:26.188]: Info: NvEnc: created encoder HEVC P1 yuv444 10-bit two-pass rfi
[2024-11-13 22:05:26.271]: Info:
[2024-11-13 22:05:26.271]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2024-11-13 22:05:26.271]: Info:
[2024-11-13 22:05:26.271]: Info: Found H.264 encoder: h264_nvenc [nvenc]
[2024-11-13 22:05:26.271]: Info: Found HEVC encoder: hevc_nvenc [nvenc]
[2024-11-13 22:05:26.288]: Info: Configuration UI available at [https://localhost:47990]
[2024-11-13 22:05:27.035]: Info: Registered Sunshine mDNS service
[2024-11-13 22:05:40.409]: Info: Encoder reenumeration is required
[2024-11-13 22:05:40.410]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024-11-13 22:05:40.410]: Info: Trying encoder [nvenc]
[2024-11-13 22:05:40.605]: Info:
Device Description : NVIDIA GeForce RTX 3080
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002216
Device Video Mem   : 10067 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1440
Offset             : 0x0
Virtual Desktop    : 2560x1440
[2024-11-13 22:05:40.606]: Info: Active GPU has HAGS disabled
[2024-11-13 22:05:40.606]: Info: Using realtime GPU priority
[2024-11-13 22:05:40.606]: Info:
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.685547,0.308594]
Green Primary      : [0.264648,0.667969]
Blue Primary       : [0.150391,0.0576172]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.2465 nits
Max Luminance      : 417.712 nits
Max Full Luminance : 417.712 nits
[2024-11-13 22:05:40.609]: Info: Desktop resolution [2560x1440]
[2024-11-13 22:05:40.609]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024-11-13 22:05:40.609]: Info: Display refresh rate [165.08Hz]
[2024-11-13 22:05:40.609]: Info: Requested frame rate [60fps]
[2024-11-13 22:05:40.609]: Info: Creating encoder [h264_nvenc]
[2024-11-13 22:05:40.609]: Info: Color coding: SDR (Rec. 601)
[2024-11-13 22:05:40.610]: Info: Color depth: 8-bit
[2024-11-13 22:05:40.610]: Info: Color range: JPEG
[2024-11-13 22:05:40.693]: Info: NvEnc: created encoder H.264 P1 two-pass rfi
[2024-11-13 22:05:40.725]: Info: Creating encoder [hevc_nvenc]
[2024-11-13 22:05:40.725]: Info: Color coding: SDR (Rec. 601)
[2024-11-13 22:05:40.726]: Info: Color depth: 8-bit
[2024-11-13 22:05:40.726]: Info: Color range: JPEG
[2024-11-13 22:05:40.808]: Info: NvEnc: created encoder HEVC P1 two-pass rfi
[2024-11-13 22:05:40.841]: Info: Creating encoder [av1_nvenc]
[2024-11-13 22:05:40.841]: Info: Color coding: SDR (Rec. 601)
[2024-11-13 22:05:40.841]: Info: Color depth: 8-bit
[2024-11-13 22:05:40.841]: Info: Color range: JPEG
[2024-11-13 22:05:40.873]: Error: NvEnc: encoding format is not supported by the gpu
[2024-11-13 22:05:40.883]: Info: Creating encoder [h264_nvenc]
[2024-11-13 22:05:40.883]: Info: Color coding: SDR (Rec. 601)
[2024-11-13 22:05:40.883]: Info: Color depth: 8-bit
[2024-11-13 22:05:40.883]: Info: Color range: JPEG
[2024-11-13 22:05:40.963]: Info: NvEnc: created encoder H.264 P1 yuv444 two-pass rfi
[2024-11-13 22:05:41.204]: Info:
Device Description : NVIDIA GeForce RTX 3080
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002216
Device Video Mem   : 10067 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1440
Offset             : 0x0
Virtual Desktop    : 2560x1440
[2024-11-13 22:05:41.205]: Info: Active GPU has HAGS disabled
[2024-11-13 22:05:41.205]: Info: Using realtime GPU priority
[2024-11-13 22:05:41.205]: Info:
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.685547,0.308594]
Green Primary      : [0.264648,0.667969]
Blue Primary       : [0.150391,0.0576172]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.2465 nits
Max Luminance      : 417.712 nits
Max Full Luminance : 417.712 nits
[2024-11-13 22:05:41.207]: Info: Desktop resolution [2560x1440]
[2024-11-13 22:05:41.208]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024-11-13 22:05:41.208]: Info: Display refresh rate [165.08Hz]
[2024-11-13 22:05:41.208]: Info: Requested frame rate [60fps]
[2024-11-13 22:05:41.208]: Info: Creating encoder [hevc_nvenc]
[2024-11-13 22:05:41.208]: Info: Color coding: SDR (Rec. 709)
[2024-11-13 22:05:41.208]: Info: Color depth: 10-bit
[2024-11-13 22:05:41.208]: Info: Color range: JPEG
[2024-11-13 22:05:41.406]: Info: NvEnc: created encoder HEVC P1 yuv444 10-bit two-pass rfi
[2024-11-13 22:05:41.481]: Info:
[2024-11-13 22:05:41.481]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2024-11-13 22:05:41.481]: Info:
[2024-11-13 22:05:41.481]: Info: Found H.264 encoder: h264_nvenc [nvenc]
[2024-11-13 22:05:41.481]: Info: Found HEVC encoder: hevc_nvenc [nvenc]
[2024-11-13 22:05:41.481]: Info: Executing [Desktop]
[2024-11-13 22:05:41.625]: Info: New streaming session started [active sessions: 1]
[2024-11-13 22:05:41.827]: Info: nvprefs: No need to modify application profile settings
[2024-11-13 22:05:41.827]: Info: nvprefs: No need to modify global profile settings
[2024-11-13 22:05:41.897]: Info: CLIENT CONNECTED
[2024-11-13 22:05:42.132]: Info: Changed virtual audio sink format to [S24 48000 2.0]
[2024-11-13 22:05:42.194]: Info: Audio capture format is [F32 48000 2.0]
[2024-11-13 22:05:42.194]: Info: Opus initialized: 48 kHz, 2 channels, 96 kbps (total), LOWDELAY
[2024-11-13 22:05:42.269]: Info:
Device Description : NVIDIA GeForce RTX 3080
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002216
Device Video Mem   : 10067 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1440
Offset             : 0x0
Virtual Desktop    : 2560x1440
[2024-11-13 22:05:42.269]: Info: Active GPU has HAGS disabled
[2024-11-13 22:05:42.269]: Info: Using realtime GPU priority
[2024-11-13 22:05:42.269]: Info:
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.685547,0.308594]
Green Primary      : [0.264648,0.667969]
Blue Primary       : [0.150391,0.0576172]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.2465 nits
Max Luminance      : 417.712 nits
Max Full Luminance : 417.712 nits
[2024-11-13 22:05:42.272]: Info: Desktop resolution [2560x1440]
[2024-11-13 22:05:42.273]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024-11-13 22:05:42.273]: Info: Display refresh rate [165.08Hz]
[2024-11-13 22:05:42.273]: Info: Requested frame rate [30fps]
[2024-11-13 22:05:42.273]: Info: Creating encoder [hevc_nvenc]
[2024-11-13 22:05:42.273]: Info: Color coding: SDR (Rec. 709)
[2024-11-13 22:05:42.273]: Info: Color depth: 8-bit
[2024-11-13 22:05:42.273]: Info: Color range: MPEG
[2024-11-13 22:05:42.308]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024-11-13 22:05:42.351]: Info: NvEnc: created encoder HEVC P1 two-pass rfi
[2024-11-13 22:07:05.509]: Info: CLIENT DISCONNECTED
[2024-11-13 22:07:12.236]: Info: Executing [Desktop]
[2024-11-13 22:07:12.287]: Info: New streaming session started [active sessions: 1]
[2024-11-13 22:07:12.461]: Info: nvprefs: No need to modify application profile settings
[2024-11-13 22:07:12.461]: Info: nvprefs: No need to modify global profile settings
[2024-11-13 22:07:12.722]: Info: CLIENT CONNECTED
[2024-11-13 22:07:12.798]: Info: Changed virtual audio sink format to [S24 48000 2.0]
[2024-11-13 22:07:12.860]: Info: Audio capture format is [F32 48000 2.0]
[2024-11-13 22:07:12.861]: Info: Opus initialized: 48 kHz, 2 channels, 96 kbps (total), LOWDELAY
[2024-11-13 22:07:13.077]: Info:
Device Description : NVIDIA GeForce RTX 3080
Device Vendor ID   : 0x000010DE
Device Device ID   : 0x00002216
Device Video Mem   : 10067 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 16347 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1440
Offset             : 0x0
Virtual Desktop    : 2560x1440
[2024-11-13 22:07:13.078]: Info: Active GPU has HAGS disabled
[2024-11-13 22:07:13.078]: Info: Using realtime GPU priority
[2024-11-13 22:07:13.078]: Info:
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.685547,0.308594]
Green Primary      : [0.264648,0.667969]
Blue Primary       : [0.150391,0.0576172]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.2465 nits
Max Luminance      : 417.712 nits
Max Full Luminance : 417.712 nits
[2024-11-13 22:07:13.082]: Info: Desktop resolution [2560x1440]
[2024-11-13 22:07:13.082]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024-11-13 22:07:13.082]: Info: Display refresh rate [165.08Hz]
[2024-11-13 22:07:13.082]: Info: Requested frame rate [30fps]
[2024-11-13 22:07:13.083]: Info: Creating encoder [hevc_nvenc]
[2024-11-13 22:07:13.083]: Info: Color coding: SDR (Rec. 709)
[2024-11-13 22:07:13.083]: Info: Color depth: 8-bit
[2024-11-13 22:07:13.083]: Info: Color range: MPEG
[2024-11-13 22:07:13.118]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024-11-13 22:07:13.171]: Info: NvEnc: created encoder HEVC P1 two-pass rfi
```

**Comment by Tmoney2014** [comment]
I have the same issue 
I try to change audio configuration, took 2 hours trying to fix it

nice catch !!

**Comment by Tmoney2014** [comment]
Basic operating principle: When streaming starts, Sunshine changes the system's default audio output device to a virtual audio device such as 'Steam Streaming Speakers'.
Issue: Applications that were running before the streaming started are likely to have already set their output to the existing audio device. In this case, these applications do not automatically switch to the new virtual audio device.
Expected behavior: Theoretically, all audio should be sent to the virtual audio device, but in practice, this is not always the case.
Possible solutions:
You may need to manually change the audio output of each application to the virtual audio device.
You can adjust the 'Audio Sink' option in Sunshine settings to specify the desired audio device.
Limitations: Currently, Sunshine cannot selectively capture or control audio from individual applications. It operates by capturing the entire system audio.
In conclusion, this issue arises from the interaction between Sunshine's current design and Windows' audio system. While there is no perfect solution yet, manually adjusting the audio output of applications or changing Sunshine's settings may help. We hope that the Sunshine development team will be able to improve this issue in the future.




Please refer to the above content.
I haven't tested it yet, but it seems to make sense, so I'm passing it on to you.
It looks like it will take about 18 hours before I can test it.

**Comment by ReenigneArcher** [comment]
We don't have any way to change the audio device of individual applications. If they are set to use the default audio device they should automatically switch over, but it may not depending on the individual program.

Unfortunately there is really nothing we can do about this.
